test/common/access_log/access_log_impl_test.cc

```diff
@@ -834,6 +834,7 @@ name: envoy.file_access_log
       RequestInfo::ResponseFlag::FaultInjected,
       RequestInfo::ResponseFlag::RateLimited,
       RequestInfo::ResponseFlag::UnauthorizedExternalService,
+      RequestInfo::ResponseFlag::RateLimitServiceError,
```

Comment:
Interesting. This test should fail due to unmatched expectation [here](https://github.com/envoyproxy/envoy/blob/37b691369b0c2a0dc64ebd80aa878ac28ca72e38/test/common/access_log/access_log_impl_test.cc#L845), because the new flag was not added to the filter config [here](https://github.com/envoyproxy/envoy/blob/37b691369b0c2a0dc64ebd80aa878ac28ca72e38/test/common/access_log/access_log_impl_test.cc#L802)

Reply:
I'm going to take a look in the morning.

Reply:
Great Catch. I see the problem. The hex value I gave for RateLimitServiceError is incorrect so the intersectResponseFlags is returning wrong value. I have corrected it.